housing_cooperative_base/models/lease.py

```diff
@@ -140,6 +150,28 @@ def _compute_state(self):
             else:
                 lease.state = "draft"
 
+    @api.multi
+    @api.depends("lease_line_ids")
+    def _compute_contains_arcade(self):
+        for lease in self:
+            premise_ids = lease.lease_line_ids.mapped("premise_id").ids
+            lease.contains_arcade = bool(
+                self.env["hc.housing"].search(
+                    [
+                        ("is_arcade", "=", True),
+                        ("premise_id", "in", premise_ids),
+                    ]
+                )
+            )
+
+    @api.onchange("signatory_ids")
+    def onchange_signatory_ids(self):
+        self.inhabitant_ids |= self.signatory_ids
+
+    @api.onchange("inhabitant_ids")
+    def onchange_inhabitant_ids(self):
+        self.inhabitant_ids |= self.signatory_ids
```

Comment:
There are probably conflicts or loops here.

Reply:
Cleaner in upcoming commits!